module/zfs/dbuf.c

```diff
@@ -2422,9 +2413,7 @@ dmu_buf_set_crypt_params(dmu_buf_t *db_fake, boolean_t byteorder,
 	dmu_buf_will_dirty_impl(db_fake,
 	    DB_RF_MUST_SUCCEED | DB_RF_NOPREFETCH | DB_RF_NO_DECRYPT, tx);
 
-	dr = db->db_last_dirty;
-	while (dr != NULL && dr->dr_txg > tx->tx_txg)
-		dr = dr->dr_next;
+	dr = dbuf_find_dirty_lte(db, tx->tx_txg);
```

Comment:
given that 2 lines down we are asserting that we found a dirty record with exactly this txg, I think this can be converted to `dbuf_find_dirty_eq`